app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerTest.kt

```diff
@@ -316,15 +313,15 @@ class BloodPressureEntrySheetControllerTest {
   @Suppress("IMPLICIT_CAST_TO_ANY")
   @Parameters(method = "params for checking valid date input")
   fun `when save is clicked, date entry is active, but input is invalid then BP measurement should not be saved`(
-      openAs: OpenAs,
-      result: UserInputDateValidator.Result
+      testParams: DoNotSaveBpWithInvalidDateTestParams
```

Comment:
For future PRs, can we have separate commits for unifying parameterized tests, and replacing mocks with real objects? It's hard to keep track of which changes are related to which change.

Reply:
I agree. I was sqashing most of the commits.